book_source/03_topical_pages/09_standalone_tools.Rmd

```diff
@@ -159,83 +159,178 @@ Processing functions currently available are,
 
 2. **Sign up for NASA Earthdata**. Using AppEEARS requires an Earthdata account visit this [page](https://urs.earthdata.nasa.gov/users/new) to create your own account.
 
-3. **Install the Python dependencies**. Using this module requires Python3 and the package manager pip to be installed in your system. 
-   To install the additional Python dependencies required, 
-a. Navigate to `pecan/modules/data.remote/inst` If you are inside the pecan directory, this can be done by,
+3. **Install the RpTools package**. Python codes required by this module are stored in a Python package named "RpTools" using this requires Python3 and the package manager pip to be installed in your system. 
+   To install the package,
+a. Navigate to `pecan/modules/data.remote/inst/RpTools` If you are inside the pecan directory, this can be done by,
 ```bash
-cd modules/data.remote/inst
+cd modules/data.remote/inst/RpTools
 ```
-b. Use pip to install the dependencies.
+b. Use pip to install the package. "-e" flag is used to install the package in an editable or develop mode, so that changes made to the code get updated in the package without reinstalling.
 ```bash
-pip install -r requirements.txt
+pip install -e .
```

Comment:
I'm not able to install this

```
$ pip install -e .
Obtaining file:///home/istfer/pecan/modules/data.remote/inst/RpTools
Collecting attrs>=19.3.0 (from RpTools==0.1)
  Using cached https://files.pythonhosted.org/packages/a2/db/4313ab3be961f7a763066401fb77f7748373b6094076ae2bda2806988af6/attrs-19.3.0-py2.py3-none-any.whl
Collecting cachetools>=4.1.1 (from RpTools==0.1)
  Could not find a version that satisfies the requirement cachetools>=4.1.1 (from RpTools==0.1) (from versions: 0.0.0, 0.1.0, 0.2.0, 0.3.0, 0.3.1, 0.4.0, 0.5.0, 0.5.1, 0.6.0, 0.7.0, 0.7.1, 0.8.0, 0.8.1, 0.8.2, 1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.1.6, 2.0.0, 2.0.1, 2.1.0, 3.0.0, 3.1.0, 3.1.1)
No matching distribution found for cachetools>=4.1.1 (from RpTools==0.1)
You are using pip version 9.0.1, however version 20.2.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

```
Looks like our python versions are out of date? When I try `pip install --upgrade pip` I get

```
Collecting pip
  Using cached https://files.pythonhosted.org/packages/5a/4a/39400ff9b36e719bdf8f31c99fe1fa7842a42fa77432e584f707a5080063/pip-20.2.2-py2.py3-none-any.whl
Installing collected packages: pip
  Found existing installation: pip 9.0.1
    Uninstalling pip-9.0.1:
Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/req/req_set.py", line 778, in install
    requirement.uninstall(auto_confirm=True)
  File "/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 754, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/usr/lib/python2.7/shutil.py", line 300, in move
    rmtree(src)
  File "/usr/lib/python2.7/shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "/usr/lib/python2.7/shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "/usr/lib/python2.7/shutil.py", line 250, in rmtree
    os.remove(fullname)
OSError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/pip-9.0.1-py2.7.egg/pip/wheel.pyc'
```

@mdietze is it possible to upgrade pecan2?

Reply:
Could you try doing `pip3 install -e .`

Reply:
```
pip3 install -e .
The program 'pip3' is currently not installed. To run 'pip3' please ask your administrator to install the package 'python3-pip'
```

Reply:
Oh ok I think you also have Python 2 installed.  You will need pip3 here.

Reply:
Yeah that's why I need @mdietze's help, I happen to set up your branch on a machine where I don't have sudo privileges, even then I would be hesitant to do a system-wide installation